### DISCUSSION TITLE
How to have an AutoRouter on a Material dialog displayed with showDialog

### DISCUSSION THREAD

**Discussion by eznibe** [Milad-Akarie/auto_route_library#1912] [Q&A]
Hi,
I would like to have an AutoRouter in the content of a dialog displayed with showDialog but I can't make it work and I can't find an example of something similar. 

I'll provide more details in case some can help me with this request.

For example, on init the route **/test** will render some text and I'll use showDialog to open a Material dialog, the content of it should have an AutoRouter where I'd like to define the nested navigation inside that dialog.
eg. **/test/subpath1** and **/test/subpath2** pages should be displayed inside my already opened dialog. The idea is to navigate to those subpaths and only refresh teh content of the dialog without changing anything in the background. Without closing and reopening the dialog.

Any idea on how to do that?

```
AutoRoute(
  path: "test",
  page: WithDialog.page,
  children: [
    AutoRoute(
      path: "subpath1",
      page: DialogContent1.page,
    ),
    AutoRoute(
      path: "subpath2",      
      page: DialogContent2.page,
    ),
  ],
),
```

**Comment by Milad-Akarie** [comment]
@eznibe try the following..
make sure you use auto_route 8.0.3 because it has a fix related to this
```dart
CustomRoute(
      page: HomeRoute.page,
      path: '/home',
      customRouteBuilder: (context, child, page) {
        return DialogRoute(
          context: context,
          builder: page.buildPage, 
          settings: page, // this's important
        );
      },
   children: ...
```

**Reply by eznibe** [reply]
thanks for the reply.

I'm new to auto_route and I don't fully understand the behavior of the CustomRoute's yet. So I have 2 questions:

. In the example you provided, when I go to **/home** it should automatically show a dialog with the content of **HomeRoute.page** inside because of the **DialogRoute** returned in the customRouteBuilder? Or I need to do something extra to show it?

. Do you think that if I add an AutoRouter() insde the content of this route with **page: HomeRoute.page** then the children of it would render inside the dialog?
